src/utils/normalization.ts

```diff
@@ -112,3 +112,31 @@ export const normalize = (payload: OperationsRepository) => {
 
   return { apis, documentation } as OperationsRepository;
 };
+
+export const emptyObject = (object: any, preserveValueKeys: string[], ignoreNestedKeys: string[]) => {
+  const processedTuples = Object.entries(object).map(([key, value]) => {
+    if (typeof value === 'object' && !ignoreNestedKeys.includes(key)) {
+      return [key, emptyObject(value, preserveValueKeys, ignoreNestedKeys)];
+    }
+
+    return [key, preserveValueKeys.includes(key) ? object[key] : emptyReturn(object[key])];
```

Comment:
You said you were having issues with Object.entries.
Object.entries returns an array of arrays. The sub-arrays are key-value pairs. They look like this:
```
[
  [key, value],
  [key, value]
]
```

 We can map over an array of key/value pairs, which is what we're doing here. We map over them and as we do we process them recursively. When we're done processing them we use `Object.fromEntries` to turn them back into an Object (which expects key/value pairs).

map returns a new object, so there's no mutation and hence the error message disappears.